### ISSUE TITLE
Not possible to select publication language after submission

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#5502]
**Describe the bug**
If a journal allows submissions in more than one language, it's possible to select the language of a submission when making the submission. This is saved to the publication object and effects which locale is required in multilingual form fields. However, once submitted, there's no way to change the publication's locale.

**To Reproduce**
1. Set up the journal to support more than one language (Settings > Website > Languages) for submissions.
2. Make a submission and select a language that is not the journal's primary language.
3. Once the submission is complete, there is no field in the publication forms to change the language.

**What application are you using?**
OJS/OMP 3.2 (`master` pre-release)

**Additional information**
It probably makes the most sense to add a language field to the metadata forms. See #5000 for related work that needs to be done on language metadata.

**Comment by asmecher** [comment]
A potentially tricky UI/UX problem needs solving here. All required submission-level fields must be completed in the submission's primary language, so a change if primary locale when that's not the case would need...
- to be prevented unless all required fields are completed in the new locale, or
- to lead the user to a form where these could be completed (though they may currently be fairly far-flung in the UI), or
- to duplicate data from the old primary locale into the new one (I don't recommend this as it would lead to data poisoning).

**Comment by NateWr** [comment]
Ok, this makes sense and the first option may be pretty easy to accomplish with the new setup. All edits are run through a validation script which can check for missing data and return an error describing which fields are missing.

I think trying to offer a form is not a good idea, especially now that we have the flexible schema so it's not necessarily very easy to construct such a form.

**Comment by pmangahis** [comment]
Will this allow JM and Editors to see what language the submission was made? We have a multi-lingual client that's accepting submission in 3 languages with no easy way to decern what language the authors have selected in the submission form.

**Comment by NateWr** [comment]
@pmangahis yes, if the UI offered a way to change the language then editors could identify which language it was submitted in. We should try to get this into 3.3 if possible.

**Comment by isgrim** [comment]
@asmecher @NateWr Hi Alec and Nate,

is there any status update for this issue?

We have problems with this in combination with the quickSubmit plugin. If you select a submission language in the quickSubmit Plugin and only want to fill the Contributor meta data for this language it is currently not possible, because it seems the Author form is not switched to the new language and the other submission language is marked as required. So you have fill at least two metadata fields.

I thought it would be no problem if we can change the submission language after submiting and correct the contributor metadata after this, but as of now its not possible.

Thanks and regards,

Nils

**Comment by isgrim** [comment]
I think I found the issue with the submission language in the quickSubmit Plugin. I opened a separate issue:
https://github.com/pkp/pkp-lib/issues/6773

These two issues work together in devious ways :grimacing:

**Comment by NateWr** [comment]
> is there any status update for this issue?

No progress yet, unfortunately!

**Comment by fgnievinski** [comment]
related (same?) issue in v2:

#742 - Permit editors to change submission locales

**Comment by NateWr** [comment]
It looks like #742 is filed against OJS 2.x. I've closed that issue because we are not going to put out an update for it.

**Comment by librariam** [comment]
+1 from a multilingual journal that uses Quick Submit and currently has no way of correcting the publication locale that was selected by mistake. This affects their metadata export to Coalition Publica that requires the title in the field of that language.

**Comment by fgnievinski** [comment]
I just wanted to make two suggestions for warning editors about the primary language:

1) the primary language's status could be emphasized including the text "primary" in brackets, for example, "Português (Brasil) [primary]".

2) a tooltip message could emphasize its importance:

> The primary language's metadata will be used when the article is exported, e.g., for DOI registration and deposit in CrossRef.

Currently the submission's primary language is formatted in bold gray by default:
![image](https://user-images.githubusercontent.com/5018500/209216364-844941f1-970d-4050-b3ba-21e34b416d7c.png)

But if you click on a secondary language, we have to language names in bold, the secondary also in color:

![image](https://user-images.githubusercontent.com/5018500/209216420-d3a4ed24-b248-491d-9b77-bf34d680fdc4.png)

And you have to click again on the secondary language to unselect it -- maybe clicking on the primary language could do the same?

**Comment by nils-stefan-weiher** [comment]
This is a major issue for us now, as we are making preparations to upgrade from 3.1.2 and 3.4.0 with multilingual submissions we need this option.

**Comment by asmecher** [comment]
@nils-stefan-weiher, can you describe some use cases for this? Why would an author choose the "wrong" primary language, and under what circumstances would someone (author, editor, etc) want to change it?

**Comment by ajnyga** [comment]
I think the use case for this is simply the author making mistakes during the submission process. Or the Editor doing a mistake when adding for example back content with Quicksubmit.

We are hoping to tackle this in CRAFT-OA project and already have some suggestions for it. I think the minimum is that we need to show the language of the submission in the UI clearly. Changing the language is more complex and we need to figure out that in which circumstances it should be allowed.

**Comment by nils-stefan-weiher** [comment]
> I think the use case for this is simply the author making mistakes during the submission process. Or the Editor doing a mistake when adding for example back content with Quicksubmit.

Its about this and about old metadata. We have some very old journals which may have incorrect metadata for their submissions. This would be helpful to correct the metadata for different languages.

**Comment by xawery79** [comment]
I would like to confirm that the restoration of the option to change the publication language is absolutely necessary. The lack of the option causes a mess not only in the metadata indexed in OJS but especially in the export to bibliographic databases. Administrators and editors are unable to correct the mistake of choosing the wrong language. In fact, the only option is to correct the entry in the locale column of the submissions table in the database.

**Comment by jyhein** [comment]
# Issue description

If a journal allows submissions in more than one language, it's possible to select the language of a submission when making the submission. This is saved to the submission object and affects which language is required in metadata form fields. However, once submitted, there's no way to change the submission’s/publication's language. The language of the submission is also completely hidden in the UI and can be only guessed by looking at the active metadata fields and content.

# Changes

- In the Submission workflow, add a button next to the 'Activity Log' button. The button shows the current language of the submission. The button opens a modal window. In the modal window, the user can change the language and is informed that the required fields need to be refilled after choosing the submission language. Picture 5 shows the window.
- Always load the submission metadata forms for the selected submission language.
- The submission language can be changed when the  submission is not published and there is only one version

# Questions
Q: We might need to consider moving locale setting from Submission to Publication (discussed with Alec)

## The attached PR's includes the following code changes:
- new form to change submission language; locale, required metadata, and author names are updated
- e.g. in workflow > publication, submission metadata forms now include submission locale
- in workflow page, new button to show current submission language and clicking it opens the modal to change language. For author's, only shows language (button diaabled)
- OJS: new test, changes submission language to French (Canada) and back to English

PRs:
PKP: https://github.com/pkp/pkp-lib/pull/9245
UI: https://github.com/pkp/ui-library/pull/280
OJS: https://github.com/pkp/ojs/pull/4016
OMP: https://github.com/pkp/omp/pull/1451
OPS: https://github.com/pkp/ops/pull/563

**Comment by asmecher** [comment]
@bozana, again, I think you're already looking at these, but just in case!
Thanks, @jyhein!

**Comment by ajnyga** [comment]
Hey @librariam @nils-stefan-weiher @pmangahis. In your experience, when is this feature usually required:
1. Before the Submission is published, ie. during the editorial process
2. Right after the Submission has been published, ie. when there is just one version
3. When there are multiple versions.

Also, we are considering how likely it would be that when an editor wants to change the language, they start by creating a new version?

The reason why we are asking is that the locale setting at the moment is stored in the Submission object. This means that if we want to keep it there, only options 1 and 2 above are possible. With option 3 we would need to move the locale setting from Submission to Publication and this will likely have a lot of implications.

**Comment by pmangahis** [comment]
The issue of being unable to see the submission languages has come up for clients when a journal accepts submissions in multiple languages and then decides to change this. They then run into the issue of being unable to edit the existing metadata information because the language of submission has been turned off. There is no way for the Editors to see this on the front end, and the Systems would be required to look in the database to see what the original language was and manually update it to allow them to update 

It would also be helpful to see this during the submission process. Particularly when trying to decide who to send the review to. We've received feedback from multilingual journals that requests have been sent to reviewers who don't speak the language of submission. It is a pretty easy mistake to make when they cannot see that the author has pre-filled the submission language and metadata in multiple languages.

**Comment by jnugent** [comment]
@ajnyga there was a bug I found earlier that is closed now where a locale column ended up on the publications table and did indeed cause problems.  https://github.com/pkp/pkp-lib/issues/7958

**Comment by ajnyga** [comment]
Thanks @pmangahis. So it seems that the issue in your case has been mostly the inability to see the submission language during the process. This is also being considered here in addition to allowing editors to change the primary locale of a submission.

For the related issue, not able to see metadata for locale that has been turned off, we are working on an improvement where the forms are _always_ accessible for the languages that have been used in the submission regardless of the settings. See https://github.com/pkp/pkp-lib/issues/9425

**Comment by MiKoko856** [comment]
Please let me know if this issue has been resolved in OJS 3.3.010.

**Comment by ajnyga** [comment]
hi @MiKoko856, no this is an open issue we are working on now.

**Comment by MiKoko856** [comment]
Hi @ajnyga, it is a great pity. Is it already possible to determine in which version of OJS this serious problem will be solved?
Is there any solution to this problem other than deleting the faulty submissions and re-entering all the metadata with only minor corrections to the title and language of the submissions?

**Comment by marcbria** [comment]
If you are in a hurry, you can edit this directly in your DB.
I can help you with this if you like. Just open a post in the support forum and mention me.

But please don't rush the developers.
There are a thousand fronts and everyone considers that their problem is more urgent than that of others.
Stressing them with our urgencies does not make their work easier.

Cheers,
m.

**Comment by asmecher** [comment]
@MiKoko856, we expect it'll be possible to change the submission language in OMP/OPS/OJS 3.5, which is the next major release. But can you describe your use case for changing the metadata? For example, is it due to a change in journal policy ("we no longer support language XYZ but want to continue existing submissions"), author error (selecting the wrong language), or something else?

**Comment by MiKoko856** [comment]
@marcbria and @asmecher, my case of changing metadata: it results from the need to correct the author's error (choosing the wrong language of the submitted text).

**Comment by jardakotesovec** [comment]
@jyhein @bozana Its merged now. If you could just help me to verify that it works as intended and I did not introduce any bug that would very helpful. Than this could be closed.

Thank you.

**Comment by bozana** [comment]
Tested and everything seems to be OK, thus closing...
